src/EditorFeatures/Core.Wpf/Preview/PreviewFactoryService.cs

```diff
@@ -494,6 +558,55 @@ public Task<object> CreateChangedAdditionalDocumentPreviewViewAsync(TextDocument
+            // will be used inside a projection buffer inside our inline diff preview below
+            // and platform's implementation currently has a bug where projection buffers
+            // are being leaked. This leak means that if we use the original buffer that is
+            // currently visible in the editor here, the projection buffer span calculation
+            // would be triggered every time user changes some code in this buffer (even though
+            // the diff view would long have been dismissed by the time user edits the code)
+            // resulting in crashes. Instead we create a new buffer from the same content.
+            // TODO: We could use ITextBufferCloneService instead here to clone the original buffer.
+            var oldBuffer = CreateNewPlainTextBuffer(oldDocument, cancellationToken);
+            var newBuffer = CreateNewPlainTextBuffer(newDocument, cancellationToken);
+
+            // Convert the diffs to be line based.  
+            // Compute the diffs between the old text and the new.
+            var diffResult = ComputeEditDifferences(oldDocument, newDocument, cancellationToken);
+
+            // Need to show the spans in the right that are different.
+            var originalSpans = GetOriginalSpans(diffResult, cancellationToken);
+            var changedSpans = GetChangedSpans(diffResult, cancellationToken);
+
+            string description = null;
+            var originalLineSpans = CreateLineSpans(oldBuffer.CurrentSnapshot, originalSpans, cancellationToken);
+            var changedLineSpans = CreateLineSpans(newBuffer.CurrentSnapshot, changedSpans, cancellationToken);
+
+            // TODO: Why aren't we attaching conflict / warning annotations here like we do for regular documents above?
+
+            // Create PreviewWorkspaces around the buffers to be displayed on the left and right
+            // so that all IDE services (colorizer, squiggles etc.) light up in these buffers.
+            var leftDocumentId = DocumentId.CreateNewId(oldDocument.Project.Id);
+            var leftSolution = oldDocument.Project.Solution
+                .RemoveAnalyzerConfigDocument(oldDocument.Id)
+                .AddAnalyzerConfigDocument(leftDocumentId, oldDocument.Name, oldBuffer.AsTextContainer().CurrentText);
+            var leftWorkspace = new PreviewWorkspace(leftSolution);
+            leftWorkspace.OpenAnalyzerConfigDocument(leftDocumentId);
+
+            var rightWorkSpace = new PreviewWorkspace(
+                oldDocument.Project.Solution.WithAnalyzerConfigDocumentText(oldDocument.Id, newBuffer.AsTextContainer().CurrentText));
+            rightWorkSpace.OpenAnalyzerConfigDocument(newDocument.Id);
+
+            return CreateChangedDocumentViewAsync(
+                oldBuffer, newBuffer, description, originalLineSpans, changedLineSpans,
+                leftWorkspace, rightWorkSpace, zoomLevel, cancellationToken);
+        }
```

Comment:
are these functions very duplicative with other functions that deal with normal/additional files?

Reply:
Yes, almost everything I have added for AnalyzerConfigDocuments in this PR is clone of the functions for additional files. We can potentially share code by using individual document specific delegates as parameters of shared method or just keep the clones. I feel @jasonmalinowski is in the best position to make this call as he is the one who more often then not deals with the code in these area. I have taken the clone approach for now, and if required we can clean this up with follow-up PRs to share more code.

Reply:
Ah yes, i believe i complained about the same thing with @jasonmalinowski about the same thing.  

For these sections we coudl at least try o have some helper shared methods...

Reply:
For huge things like this I'd expect that we'd unify the code. If there's some small bits that can't quite make it unifiable like the add/remove or Open methods, my pattern was to unify to a helper that took lambdas that did those few bits. I think doing that also made it clear where the "differences" were and how we can then better unify code in the future.

Reply:
Sounds good - I will update the PR to to share code for larger methods.

Reply:
I ended up doing both the changes, as there seem to be lot of places where we pass the document kind as a boolean flag, and an enum is an obvious choice instead of multiple bools. I have kept the enum as internal for now, but we can make it public in future if we believe it is required and approved by the design team.